source/gui/nvdaControls.py

```diff
@@ -376,6 +377,56 @@ def _onShowEvt(self, evt):
 		evt.Skip()
 
 
+class ContinueCancelDialog(MessageDialog):
+	"""
+	This implementation of a `gui.nvdaControls.MessageDialog`, provides `Continue` and `Cancel` buttons as its controls.
+	These serve the same functions as `OK` and `Cancel` in other dialogs, but may be more desirable in some situations.
+	"""
+
+	def __init__(
+		self,
+		parent: wx.Frame,
+		title: str,
+		message: str,
+		dialogType: int = MessageDialog.DIALOG_TYPE_STANDARD,
+		continueButtonFirst: bool = True,
+	) -> None:
+		"""Creates a ContinueCancelDialog MessageDialog.
+
+		:param parent: The parent window for the dialog, usually `gui.mainFrame`.
+		:param title: The title or caption of the dialog.
+		:param message: The message to be shown in the dialog.
+		:param dialogType: One of the dialog type constants from MessageDialog, defaults to standard.
+		:param continueButtonFirst: If True, the Continue button will appear first, and be selected when the dialog
+			opens; if False, the Cancel button will. Defaults to True.
+		"""
+		self.continueButtonFirst: bool = continueButtonFirst
+		super().__init__(parent, title, message, dialogType)
+
+	def _addButtons(self, buttonHelper) -> None:
```

Comment:
please add a type hint for buttonHelper for this function and the internal functions below